lib/disk/modules/VMWareSparseDisk.rb

```diff
@@ -139,16 +189,67 @@ def d_size
 
   private
 
+  def initialize_without_markers
+    #
+    # Seek to start of the grain directory.
+    #
+    @vmwareSparseDisk_file.seek(@sparseHeader.gdOffset * blockSize, IO::SEEK_SET)
+
+    #
+    # Read the first grain directory entry to get the offset to the start of
+    # the grain tables.
+    #
+    buf = @vmwareSparseDisk_file.read(BYTES_PER_SECTOR)
+    @grainDirectory = []
+
+    GDES_PER_GD.times do |index|
+      @grainDirectory[index] = buf[index, GDE_SIZE].unpack1('L')
+    end
```

Comment:
This can be shortened to 

```ruby
    @grainDirectory = GDES_PER_GD.times.map { |index| buf[index, GDE_SIZE].unpack1('L') }
```

Reply:
You can also unpack it in a single shot, which might be more efficient...

```ruby
    @grainDirectory = buf.unpack("L{GDES_PER_GD}")
```

Reply:
since GDE_SIZE is a constant with a value of 4 (i.e. not a `long`), went with:
```ruby
@grainDirectory = GDES_PER_GD.times.map { |index| buf[index, GDE_SIZE].unpack1('L') }
```